gpflow/likelihoods/likelihoods.py

```diff
@@ -207,17 +212,41 @@ def predict_density(self, Fmu, Fvar, Y):
         :param Fmu: mean function evaluation Tensor, with shape [..., latent_dim]
         :param Fvar: variance of function evaluation Tensor, with shape [..., latent_dim]
         :param Y: observation Tensor, with shape [..., observation_dim]:
-        :return predicted density, with shape [...]
+        :return log predicted density, with shape [...]
         """
         tf.debugging.assert_equal(tf.shape(Fmu), tf.shape(Fvar))
         self._check_last_dims_valid(Fmu, Y)
-        res = self._predict_density(Fmu, Fvar, Y)
+        res = self._predict_log_density(Fmu, Fvar, Y)
         self.check_return_shape(res, Fmu, Y)
         return res
 
-    def _predict_density(self, Fmu, Fvar, Y):
+    @abc.abstractmethod
+    def _predict_log_density(self, Fmu, Fvar, Y):
         raise NotImplementedError
 
+    def predict_density(self, Fmu, Fvar, Y):
+        r"""
+        Given a Normal distribution for the latent function, and a datum Y,
+        compute the predictive density of Y.
+
+        i.e. if
+            q(F) = N(Fmu, Fvar)
+
+        and this object represents
+
+            p(y|F)
+
+        then this method computes the predictive density
+
+            ∫ p(y=Y|F)q(F) df
+
+        :param Fmu: mean function evaluation Tensor, with shape [..., latent_dim]
+        :param Fvar: variance of function evaluation Tensor, with shape [..., latent_dim]
+        :param Y: observation Tensor, with shape [..., observation_dim]:
+        :return predicted density, with shape [...]
+        """
+        return tf.exp(self.predict_log_density(Fmu, Fvar, Y))
```

Comment:
This is dangerous. In GPflow 1, predict_density DOES return the *log*-density. I don't think we should provide a method with the same name & interface but different behaviour.
```suggestion
        warnings.warn(
            "predict_density is deprecated, use predict_log_density instead",
            DeprecationWarning
        )
        return self.predict_log_density(Fmu, Fvar, Y)
```
(and add `import warnings` to the top of the file)